frontend/src/routes/ApplicationPortal.js

```diff
@@ -13,68 +13,39 @@ import LoadingBall from "src/components/LoadingBall";
 
 import * as Sentry from "@sentry/browser";
 import NavBar from "src/components/NavBar";
-
-class ApplicationPortal extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      admission: null,
-      results: undefined,
-      committees: [],
-      error: null,
-      isFetching: true,
-      myApplication: undefined,
-      selectedCommittees: {},
-      user: null,
-      isEditingApplication: true,
-    };
-  }
-
-  toggleCommittee = (name) => {
-    this.setState(
-      (state) => ({
-        selectedCommittees: {
-          ...state.selectedCommittees,
-          [name.toLowerCase()]: !state.selectedCommittees[name],
-        },
-      }),
-      () => {
-        this.persistState();
-      }
-    );
+import { useLocation } from "react-router-dom";
+
+const ApplicationPortal = () => {
+  const [admission, setAdmission] = useState(null);
+  const [results] = useState(undefined);
+  const [committees, setCommittees] = useState([]);
+  const [error, setError] = useState(null);
+  const [isFetching, setIsFetching] = useState(true);
+  const [myApplication, setMyApplication] = useState(undefined);
+  const [selectedCommittees, setSelectedCommittees] = useState({});
+  const [user, setUser] = useState(null);
+  const [isEditingApplication, setIsEditingApplication] = useState(true);
```

Comment:
I love how this implements ALL the different empty values we have. `null`, `undefined`, `[]`, `{}`. Of course not your fault 👍🏻